src/routes/event-match.tsx

```diff
@@ -91,28 +94,58 @@ const EventMatch = ({ eventKey, matchKey }: Props) => {
     [match],
   )
 
+  if (!isData(match)) {
+    return (
+      <Page
+        back={`/events/${eventKey}`}
+        name={
+          (m === null
+            ? matchKey + ' - '
+            : m.group + (m.num ? ' Match ' + m.num : '') + ' - ') +
+          (event ? event.name : eventKey)
+        }
+        class={matchStyle}
+      >
+        <Card class={matchStyle}>
+          <Icon icon={alert} />
+          <div
+            class={css`
+              font-size: 2rem;
+              text-align: center;
+            `}
+          >
+            {m === null
+              ? 'This Match Does Not Exist'
+              : m.group + (m.num ? ' Match ' + m.num : '') + ' Does Not Exist'}
```

Comment:
Can we differentiate between a network error and a server error and a 404?

If I block the request to the peregrine API in devtools and clear the cache, then it shows "Qual 1 does not exist", but can we change it to say "Unable to load Qual 1" in the case that there was a network error in fetching it?

Reply:
From what I see, if the whole thing isn't throwing a huge error, it has to be a 404. It has to get the matchKey from somewhere to run at all. If it has the matchKey and still return null, it's a 404 error; if it doesn't have the matchKey, it's probably a network or server error, but I don't think I could run anything in here to detect that.

Reply:
Never mind, I was looking in the wrong place. All the way up at L97, it's checking whether or not match is defined. I spent a couple hours trying to understand what would allow match to be defined or undefined, and I again don't understand the software quite well enough to figure that out.

Reply:
I attempted to address this in my new commit. As I mentioned, I couldn't test it because of weird errors I didn't create.